agent/attachment.go

```diff
@@ -0,0 +1,103 @@
+package agent
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/docker/swarmkit/api"
+	"github.com/docker/swarmkit/identity"
+	"github.com/docker/swarmkit/log"
+	"golang.org/x/net/context"
+)
+
+// Listener allow to receive notifications about non executor attachment objects.
+type Listener interface {
+	// Notify notifies the listener about the updates over the executor attachment objects.
+	Notify([]*api.ExecutorAttachment)
+}
+
+// ExecutorAttachmentManager provides control over executor attachments on this node.
+type ExecutorAttachmentManager interface {
```

Comment:
Why is this an interface? There is only one implementation and this is not java.